unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -206,6 +207,198 @@ public UTF8String toLowerCase() {
     return fromString(toString().toLowerCase());
   }
 
+  /**
+   * Copy the bytes from the current UTF8String, and make a new UTF8String.
+   * @param start the start position of the current UTF8String in bytes.
+   * @param end the end position of the current UTF8String in bytes.
+   * @return a new UTF8String in the position of [start, end] of current UTF8String bytes.
+   */
+  private UTF8String copyUTF8String(byte[] bytes, int start, int end) {
```

Comment:
+1, this function will create a new object, can we only modify the `base` or (`offset` and `size`)?